src/main/kotlin/net/atos/zac/signalering/SignaleringenJob.kt

```diff
@@ -0,0 +1,364 @@
+            )
+        LOG.info(
+            """Zaak signaleringen verzenden: gestopt (${info.streefdatumVerzonden} streefdatum waarschuwingen,
+            | ${info.fataledatumVerzonden} fatale datum waarschuwingen)
+            """.trimMargin()
+        )
+    }
+
+    /**
+     * Send the E-Mail warnings about the einddatum gepland
+     *
+     * @return the number of E-Mails sent
+     */
+    fun zaakEinddatumGeplandVerzenden(zaaktype: ZaakType, venster: Int): Int {
+        val verzonden = IntArray(1)
+        zoekenService.zoek(getZaakSignaleringTeVerzendenZoekParameters(DatumVeld.ZAAK_STREEFDATUM, zaaktype, venster))
+            .items.stream()
+            .map { it as ZaakZoekObject? }
+            .map {
+                buildZaakSignalering(
+                    getZaakSignaleringTarget(it, SignaleringDetail.STREEFDATUM),
+                    it,
+                    SignaleringDetail.STREEFDATUM
+                )
+            }
+            .filter { Objects.nonNull(it) }
+            .forEach { verzonden[0] += verzendZaakSignalering(it) }
+        return verzonden[0]
+    }
+
+    /**
+     * Send the E-Mail warnings about the uiterlijke einddatum afdoening
+     *
+     * @return the number of E-Mails sent
+     */
+    fun zaakUiterlijkeEinddatumAfdoeningVerzenden(
+        zaaktype: ZaakType,
+        venster: Int
+    ): Int {
+        val verzonden = IntArray(1)
+        zoekenService.zoek(
+            getZaakSignaleringTeVerzendenZoekParameters(
+                DatumVeld.ZAAK_FATALE_DATUM,
+                zaaktype,
+                venster
+            )
+        )
+            .items.stream()
+            .map { it as ZaakZoekObject? }
+            .map {
+                buildZaakSignalering(
+                    getZaakSignaleringTarget(it, SignaleringDetail.FATALE_DATUM),
+                    it,
+                    SignaleringDetail.FATALE_DATUM
+                )
+            }
+            .filter { obj: Signalering? -> Objects.nonNull(obj) }
+            .forEach { signalering: Signalering? -> verzonden[0] += verzendZaakSignalering(signalering) }
+        return verzonden[0]
+    }
+
+    fun getZaakSignaleringTarget(zaak: ZaakZoekObject?, detail: SignaleringDetail): String? {
+        if (signaleringenService.readInstellingenUser(
+                SignaleringType.Type.ZAAK_VERLOPEND,
+                zaak!!.behandelaarGebruikersnaam
+            ).isMail &&
+            !signaleringenService.findSignaleringVerzonden(
+                getZaakSignaleringVerzondenParameters(
+                    zaak.behandelaarGebruikersnaam, zaak.uuid,
+                    detail
+                )
+            ).isPresent
+        ) {
+            return zaak.behandelaarGebruikersnaam
+        }
+        return null
+    }
+
+    fun buildZaakSignalering(
+        target: String?,
+        zaakZoekObject: ZaakZoekObject?,
+        detail: SignaleringDetail
+    ): Signalering? {
+        if (target != null) {
```

Comment:
When I converted to Kotlin I removed these explicit null checks of parameters and make the function/method argument non-nullable and then in the callers of this function I would first check if the field is null or not. If it is null it does not make any sense to call this function since it will return null in that case anyway..

Also fine to leave this for now for a future PR btw!

Reply:
Filtered out objects that would result in target being null